libraries/ms-common/src/stm32f0xx/i2c.c

```diff
@@ -0,0 +1,129 @@
+#include "i2c.h"
+#include <stdbool.h>
+#include "stm32f0xx.h"
+#include "log.h"
+
+// Arbitrary timeout
+#define I2C_TIMEOUT 1000000
+#define I2C_TIMEOUT_WHILE_FLAG(i2c, flag, status) \
+do { \
+  uint32_t timeout = (I2C_TIMEOUT); \
```

Comment:
Isn't ``timeout`` is kind of misleading since it's more the number of retries and not a time interval?